src/main.ts

```diff
@@ -2,28 +2,111 @@ import * as core from '@actions/core'
 import * as io from '@actions/io'
 import * as exec from '@actions/exec'
 import * as utils from './utils'
-import {Inputs, createPullRequest} from './github-helper'
+import {
+  Inputs,
+  createPullRequest,
+  buildBranchesFromLabels
+} from './github-helper'
 
 const CHERRYPICK_EMPTY =
   'The previous cherry-pick is now empty, possibly due to conflict resolution.'
 
+export type ExecutionStatus = {
+  failed: boolean
+  branch: string
+  inputs?: Inputs
+  msg?: string | undefined | void
+}
+
+export type Statuses = {
+  completedCherryPicks: Array<ExecutionStatus>
+  cherryPickErrors: Array<ExecutionStatus>
+}
+
 export async function run(): Promise<void> {
-  try {
-    const inputs: Inputs = {
-      token: core.getInput('token'),
-      committer: core.getInput('committer'),
-      author: core.getInput('author'),
-      branch: core.getInput('branch'),
-      labels: utils.getInputAsArray('labels'),
-      assignees: utils.getInputAsArray('assignees'),
-      reviewers: utils.getInputAsArray('reviewers'),
-      teamReviewers: utils.getInputAsArray('teamReviewers')
+  const inputs: Inputs = {
+    token: core.getInput('token'),
+    committer: core.getInput('committer'),
+    author: core.getInput('author'),
+    branch: core.getInput('branch'),
+    labels: utils.getInputAsArray('labels'),
+    assignees: utils.getInputAsArray('assignees'),
+    reviewers: utils.getInputAsArray('reviewers'),
+    teamReviewers: utils.getInputAsArray('teamReviewers'),
+    allowUserToSpecifyBranchViaLabel:
+      core.getInput('allowUserToSpecifyBranchViaLabel') || '',
+    labelPatternRequirement: core.getInput('labelPatternRequirement'),
+    userBranchPrefix: core.getInput('userBranchPrefix') || ''
+  }
+
+  const branchesToCherryPick =
+    inputs.allowUserToSpecifyBranchViaLabel === 'true'
+      ? buildBranchesFromLabels(inputs)
+      : [inputs.branch]
+  core.info(`branches to cherry pick ${JSON.stringify(branchesToCherryPick)}`)
+
+  if (!branchesToCherryPick) {
+    core.info(`No branches to cherry pick`)
+    return
+  }
+
+  if (branchesToCherryPick[0] === '') {
+    core.info(`No branches to cherry pick`)
+    return
+  }
```

Comment:
```suggestion
  if (!branchesToCherryPick || branchesToCherryPick[0] === '') {
    core.info(`No branches to cherry pick`)
    return
  }
  
  core.info(`branches to cherry pick ${JSON.stringify(branchesToCherryPick)}`)
```

Reply:
I don't think we can do `branchesToCherryPick[0] === ''` if `branchesToCherryPick` is `undefined`. That's why I put that return statement after the undefined handle block. 
Let me know I'm wrong and we can condense it!

Reply:
I think this is safe since OR expressions exit early on the first operand evaluation to true. https://javascript.info/logical-operators

Reply:
Nice! Definitely not possible in python. Thanks!